schema/schema.go

```diff
@@ -67,12 +68,31 @@ type StateTree struct {
 // Base64EncodedBytes is a base64-encoded binary value.
 type Base64EncodedBytes []byte
 
-// Preconditions contain a representation of VM state at the beginning of the test
+// ChainHead represents a head tipset.
+type ChainHead []cid.Cid
+
+// PreconditionsBlockSeq are the preconditions for a blockseq vector.
+type PreconditionsBlockSeq struct {
+	GenesisTs time.Time  `json:"genesis_ts,omitempty"`
+	ChainHead *ChainHead `json:"chain_head,omitempty"`
+}
+
+// Preconditions contain the environment that needs to be set before the
+// vector's applies are applied.
 type Preconditions struct {
+	*PreconditionsBlockSeq
+
 	// Epoch must be interpreted by the driver as an abi.ChainEpoch in Lotus, or
 	// equivalent type in other implementations.
-	Epoch     int64      `json:"epoch"`
-	StateTree *StateTree `json:"state_tree"`
+	Epoch     int64      `json:"epoch,omitempty"`
+	StateTree *StateTree `json:"state_tree,omitempty"`
+
+	// CircSupply is optional. If specified, it is the value that will be
+	// injected in the VM when running this vector. If absent, the default
+	// value will be injected (TotalFilecoin, the maximum supply of Filecoin
+	// that will ever exist). It is usually odd to set it, and it's only here
+	// for specialized vectors.
+	CircSupply *int64 `json:"circ_supply,omitempty"`
```

Comment:
is it worth having this as a map from epoch to supply?

Reply:
Nah, not worth it for what the circulating supply does (after researching it).